app/Http/Controllers/Chat/ChatController.php

```diff
@@ -143,7 +144,24 @@ public function updates()
 
         $messages = $messages->get()->reverse();
 
-        if ($messages->isEmpty()) {
+        $silenceQuery = UserAccountHistory::bans()->limit(100);
+        $lastHistoryId = get_int($params['since_history'] ?? null);
+
+        if ($lastHistoryId === null) {
+            $previousMessage = Message::find($since);
```

Comment:
what's the reason for not using `$messages->last()` (first?)

Reply:
because there might be bans before the earliest current latest message?